### PR TITLE
feat: admin edit user badges

### LINKED ISSUE #2131: Add ability for admin to assign badges Frontend
### Description

The current way to assign badges for an admin is to do it directly in the Firebase Backend. This is complex and risky. We want to add this ability to the user profiles so admins can do it on the frontend.

Currently we use `badges` for `verified` users. But we are introducing a new one for `supporters` in https://github.com/ONEARMY/community-platform/issues/2130

### Page related
Will be used one user edit page(admin only) : https://dev.onearmy.world/u/username/edit

### Mockup
admin badges

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

- chore: refactor to introduce UserSettings page
- feat: adds admin panel to edit user badges

Outstanding:

- [x] Tests
- [ ] Refactor
- [x] Update user object
- [x] Limit to only allow admins access

## Git Issues

Closes #2131

## Screenshots/Videos

Following screenshot using the standard `red2` colour token. 

Screenshot 2023-02-26 at 22 10 45

### CODE REVIEW

**Comment by thisislawatts** [comment]
@davehakkens, take a look. Iterating slightly on the designs, the implementation has a standalone "Update badges" button rather than hooking into the main "Save profile" form. The aim is to keep this admin only logic separate from the normal edit profile flow.

**Comment by davehakkens** [comment]
nice @thisislawatts. Yeh works for me like this!

**Comment by AlfonsoGhislieri** [comment]
I can take a look at this one next week!

File: src/pages/Settings/UserBadgeSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+export const UserBadgeSettings = observer((props: { userId: string }) => {
+  const { userStore } = useCommonStores().stores
+  const [badges, setBadges] = useState({
+    verified: false,
+    supporter: false,
+  })
+  const [isLoading, setLoading] = useState(true)
+  const [isSaving, setSaving] = useState(false)
+
+  async function fetchUser() {
```
Comment (AlfonsoGhislieri) [inline src/pages/Settings/UserBadgeSettings.tsx:20]: Could change this to be an arrow function to be in line with the other functions

File: src/stores/User/user.store.ts [hunk 2]
```diff
@@ -151,6 +151,21 @@ export class UserStore extends ModuleStore {
     return lookup2[0]
   }
 
+  public async updateUserBadge(userId: string, badges: IUserBadges) {
+    console.log(`updateUserBadge`, badges)
+    const dbRef = this.db.collection<IUserPP>(COLLECTION_NAME).doc(userId)
+
+    console.log(`updateUserBadge.before:`, toJS(await dbRef.get('server')))
```
Comment (AlfonsoGhislieri) [inline src/stores/User/user.store.ts:158]: Are these console.log's necessary?

File: src/stores/User/user.store.ts [hunk 3]
```diff
@@ -162,6 +177,7 @@ export class UserStore extends ModuleStore {
     adminEditableUserId?: string,
   ) {
     this.setUpdateStaus('Start')
+    console.log(`updateUserProfile`, { values })
```
Comment (AlfonsoGhislieri) [inline src/stores/User/user.store.ts:180]: Is this console log also needed?

File: src/stores/User/user.store.ts [hunk 4]
```diff
@@ -194,6 +210,7 @@ export class UserStore extends ModuleStore {
     }
 
     // update on db and update locally (if targeting self as user)
+    console.log(`updateUserProfile.doc.set`, { updatedUserProfile })
```
Comment (AlfonsoGhislieri) [inline src/stores/User/user.store.ts:213]: Same as above

**Review by AlfonsoGhislieri — APPROVED** [review]
@thisislawatts looks good to me, some rogue console logs which I'm sure if are necessary though

File: src/stores/User/user.store.ts (hunk 2 above)
Comment (thisislawatts) [reply src/stores/User/user.store.ts:158]: Nope, let me remove them.

File: src/stores/User/user.store.ts (hunk 4 above)
Comment (thisislawatts) [reply src/stores/User/user.store.ts]: Removed

**Comment by thisislawatts** [comment]
Thanks for the review @AlfonsoGhislieri, I think you flagged an interesting point here around `console.log` calls, I have raised https://github.com/ONEARMY/community-platform/pull/2148, which aims to add automated checks to prevent `console` calls being used in the primary application.

File: src/pages/Settings/UserBadgeSettings.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Settings/UserBadgeSettings.tsx:20]: @AlfonsoGhislieri it is a unclear whether this is something you are asking or stating. If this is a suggestion or issue that would block you from approving this PR I recommend being explicit. 

I like to use the [Conventional Comment pattern](https://conventionalcomments.org/) although I am a little inconsistent with it.

For example your comment could adopt either of these prefixes which would help the PR author understand you intent. 

> **nitpick (non-blocking):** Could change this to be an arrow function to be in line with the other functions

> **issue (blocking):** Please change this to be an arrow function to be in line with the other functions

Comment (AlfonsoGhislieri) [reply src/pages/Settings/UserBadgeSettings.tsx:20]: Ah this is great, thanks for the resource! Updated the comment, was more of a suggestion, not a blocking issue

**Comment by AlfonsoGhislieri** [comment]
> Thanks for the review @AlfonsoGhislieri, I think you flagged an interesting point here around `console.log` calls, I have raised #2148, which aims to add automated checks to prevent `console` calls being used in the primary application.

That's a great idea, I've definitely noticied a few console.log's that snuck into the production code

File: src/pages/Settings/UserBadgeSettings.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Settings/UserBadgeSettings.tsx:20]: It's a good idea though, I have opened https://github.com/ONEARMY/community-platform/pull/2149 to see if we can introduce an eslint plugin to enforce arrow style functions. Up until now the project has not had an opinion about this convention. I don't have a preference here but like the opportunity to make the code base more consistent. 

Please take a look over the changes there and add a comment or 👍🏻 if it looks good to you.

**Review by davehakkens — APPROVED** [review]

**Pull request merged**